akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectiveExtendedExampleSpec.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package docs.http.scaladsl.server.directives
+
+import akka.event.{LoggingAdapter, Logging}
+import akka.event.Logging.LogLevel
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.server.RouteResult.Complete
+import akka.http.scaladsl.server.directives.{LoggingMagnet, LogEntry, DebuggingDirectives}
+import docs.http.scaladsl.server.RoutingSpec
+import Console._
+
+//This example shows how to log client request and response time using DebuggingDirective
+
+class DebuggingDirectiveExtendedExampleSpec extends RoutingSpec {
+
+  "logRequestResultWithResponseTime" in {
+
+    val THRESHOLD_VALUE = 3
+    def akkaResponseTimeLoggingFunction(loggingAdapter: LoggingAdapter,
+                                        requestTimestamp: Long,
+                                        level: LogLevel = Logging.InfoLevel)(req: HttpRequest)(res: Any): Unit = {
+      val entry = res match {
+        case Complete(resp) =>
+          val responseTimestamp: Long = System.currentTimeMillis()
+          val elapsedTime: Long = responseTimestamp - requestTimestamp
+          val loggingString = "Logged Request:" + req.method + ":" + req.uri + ":" + resp.status + ":" + elapsedTime
+          // This is for displaying the color of the request according to the 
+          // response time of that request.( i.e if responseTime > THRESHOLD_VALUE then it is RED or else GREEN)
+         val coloredLoggingString = if (elapsedTime > THRESHOLD_VALUE) {
+            RED + loggingString + RESET
+          } else {
+           GREEN + loggingString + RESET
+          }
```

Comment:
I think we could skip the coloring, that is a _very_ unusual requirement for http request logging